node/src/reactor/main_reactor/control.rs

```diff
@@ -371,31 +243,34 @@ impl MainReactor {
                                 )
                             }
                             Ok(None) if self.switch_block.is_none() => {
-                                if let ActivationPoint::Genesis(timestamp) =
-                                    self.chainspec.protocol_config.activation_point
-                                {
-                                    let diff = timestamp.saturating_diff(Timestamp::now());
-                                    if diff > TimeDiff::default() {
+                                if let Some(timestamp) = self.is_genesis() {
+                                    let is_validator = self.should_commit_genesis();
+                                    if false == is_validator {
+                                        return CatchUpInstruction::Fatal(
+                                            "CatchUp: only validating nodes may participate in genesis; cannot proceed without trusted hash".to_string(),
```

Comment:
Why not just create the genesis block even if we aren't a validator? Then we can use it as the trusted block.